pkg/artifactory/repositories.go

```diff
@@ -1271,199 +1267,52 @@ func isSelectRandom(opts ...bool) bool {
-		"conda":     "simple-default",
-		"debian":    "simple-default",
-		"docker":    "simple-default",
-		"gems":      "simple-default",
-		"generic":   "simple-default",
-		"gitlfs":    "simple-default",
-		"go":        "go-default",
-		"gradle":    "maven-2-default",
-		"helm":      "simple-default",
-		"ivy":       "ivy-default",
-		"maven":     "maven-2-default",
-		"npm":       "npm-default",
-		"nuget":     "nuget-default",
-		"opkg":      "simple-default",
-		"p2":        "simple-default",
-		"pub":       "simple-default",
-		"puppet":    "puppet-default",
-		"pypi":      "simple-default",
-		"sbt":       "sbt-default",
-		//"vagrant":   "simple-default", //currently not available
-		"vcs": "simple-default", //Type is only available in Remote Repository
-		"rpm": "simple-default",
-	}
-
-	if selectRandom {
-		repoLayout = selectRandomFromMapOfStrings(remoteDefaultRepoLayoutMap)
-	} else {
-		if value, ok := remoteDefaultRepoLayoutMap[repoType]; ok {
-			repoLayout = value
-		}
-	}
-	return repoLayout
+type DefaultRepoLayoutStruct struct {
+	RepoLayoutRef      string
+	SupportedRepoTypes map[string]bool
 }
 
-//Return the default repo layout of Virtual Repository type
-func getDefaultVirtualRepoLayoutRef(repoType string, opts ...bool) string {
-
-	selectRandom := isSelectRandom(opts...)
-	const defaultRepoLayout = "simple-default"
-	repoLayout := defaultRepoLayout
-
-	virtualDefaultRepoLayoutMap := map[string]string{
-		"alpine": "simple-default",
-		"bower":  "bower-default",
-		"cran":   "simple-default",
-		//"cargo":     "simple-default", //currently not available
-		"chef": "simple-default",
-		//"cocoapods": "simple-default", //currently not available
-		"composer": "composer-default",
-		"conan":    "conan-default",
-		"conda":    "simple-default",
-		"debian":   "simple-default",
-		"docker":   "simple-default",
-		"gems":     "simple-default",
-		"generic":  "simple-default",
-		"gitlfs":   "simple-default",
-		"go":       "go-default",
-		"gradle":   "maven-2-default",
-		"helm":     "simple-default",
-		"ivy":      "ivy-default",
-		"maven":    "maven-2-default",
-		"npm":      "npm-default",
-		"nuget":    "nuget-default",
-		"opkg":     "simple-default",
-		"p2":       "simple-default",
-		"pub":      "simple-default",
-		"puppet":   "puppet-default",
-		"pypi":     "simple-default",
-		"sbt":      "sbt-default",
-		//"vagrant":   "simple-default", //currently not available
-		"rpm": "simple-default",
-	}
-
-	if selectRandom {
-		repoLayout = selectRandomFromMapOfStrings(virtualDefaultRepoLayoutMap)
-	} else {
-		if value, ok := virtualDefaultRepoLayoutMap[repoType]; ok {
-			repoLayout = value
-		}
-	}
-	return repoLayout
+//Consolidated list of Default Repo Layout for all Package Types with active Repo Types
+var defaultRepoLayoutMap = map[string]DefaultRepoLayoutStruct{
+	"alpine":    {RepoLayoutRef: "simple-default", SupportedRepoTypes: map[string]bool{"local": true, "remote": true, "virtual": true, "federated": true}},
```

Comment:
There are no `false` value in the `SupportedRepoTypes` why not just a list of repo classes and then check if the requested class is in the list?

Reply:
Default is false. For a list I need to traverse and match every time. With current implementation it is a straight lookup by key.